### PR TITLE
Upgrade to Micronaut 2

### PR DESCRIPTION
**Pull request by ilopmar**
See https://github.com/micronaut-projects/micronaut-core/issues/3946

### CODE REVIEW

**Review by alvarosanchez — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -15,10 +15,6 @@ subprojects { Project subproject ->
 
     apply plugin: "io.micronaut.build.common"
 
-    if (subproject.parent.name == "examples" || subproject.name == "examples") {
```
Comment (sdelamo) [inline build.gradle:18]: why did you remove this?

Comment (ilopmar) [reply build.gradle:18]: We've fixed that in the build plugin (in version 2.0.10), so that's not necessary anymore. See https://github.com/micronaut-projects/micronaut-build/commit/07b957785beb237b5ed7a728d12b0b9262380987

Comment (ilopmar) [reply build.gradle:18]: We've also updated the project template so new projects don't need to use have it.

**Review by sdelamo — APPROVED** [review]
This should go into a mayor release of the module. Since it upgrades to Groovy 3 and Micronaut 2.

**Comment by ilopmar** [comment]
> This should go into a mayor release of the module. Since it upgrades to Groovy 3 and Micronaut 2.

Yes, it will be 2.0.0: https://github.com/micronaut-projects/micronaut-graphql/pull/103/files#diff-503f218d646c10f484fdc9d6315bf2e3R1
